src/ray/raylet/worker_pool.cc

```diff
@@ -153,10 +153,19 @@ void WorkerPool::SetAgentManager(std::shared_ptr<AgentManager> agent_manager) {
   agent_manager_ = agent_manager;
 }
 
+inline void WorkerPool::PopWorkerCallbackAsync(const PopWorkerCallback &callback,
+                                               std::shared_ptr<WorkerInterface> worker,
+                                               Status status) {
+  if (callback) {
+    // Call back this function asynchronously to make sure executed in different stack.
+    io_service_->post([callback, worker, status]() { callback(worker, status); });
```

Comment:
Don't we need to check the return value of the callback? Otherwise this seems like we could leak a worker (ClusterTaskManager isn't using the worker, but the worker is still marked as occupied).

Reply:
Can we also make sure to add a unit test for this case?

Reply:
Great！My bug. Let me fix.